#! /bin/sh /usr/share/dpatch/dpatch-run
## init.d.dpatch by Julien Danjou <acid@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch init.d scripts

@DPATCH@
diff -ru xen-3.0.1.orig/tools/examples/init.d/xend xen-3.0.1/tools/examples/init.d/xend
--- xen-3.0.1.orig/tools/examples/init.d/xend	2006-01-31 17:09:20.000000000 +0100
+++ xen-3.0.1/tools/examples/init.d/xend	2006-03-21 18:27:18.000000000 +0100
@@ -7,7 +7,9 @@
 # chkconfig: 2345 98 01
 # description: Starts and stops the Xen control daemon.
 
-if ! [ -e /proc/xen/privcmd ]; then
+XENSTORED_RUN_DIR="/var/run/xenstored"
+
+if [ ! -e /proc/xen/privcmd -a -x /usr/sbin/xend ]; then
 	exit 0
 fi
 
@@ -27,6 +29,7 @@
 
 case "$1" in
   start)
+	[ -d "$XENSTORED_RUN_DIR" ] || mkdir -p "$XENSTORED_RUN_DIR"
 	xend start
 	await_daemons_up
 	;;
diff -ru xen-3.0.1.orig/tools/examples/init.d/xendomains xen-3.0.1/tools/examples/init.d/xendomains
--- xen-3.0.1.orig/tools/examples/init.d/xendomains	2006-01-31 17:09:20.000000000 +0100
+++ xen-3.0.1/tools/examples/init.d/xendomains	2006-03-21 18:27:54.000000000 +0100
@@ -30,7 +30,7 @@
 
 # Correct exit code would probably be 5, but it's enough 
 # if xend complains if we're not running as privileged domain
-if ! [ -e /proc/xen/privcmd ]; then
+if [ ! -e /proc/xen/privcmd -a -x /usr/sbin/xm ]; then
 	exit 0
 fi
 
@@ -474,7 +474,7 @@
     restart)
 	restart
 	;;
-    reload)
+    reload|force-reload)
 	reload
 	;;
 
@@ -492,7 +492,7 @@
 	;;
 
     *)
-	echo "Usage: $0 {start|stop|restart|reload|status}"
+	echo "Usage: $0 {start|stop|restart|reload|force-reload|status}"
 	rc_failed 3
 	rc_status -v
 	;;
